backend/tournesol/views/user.py

```diff
@@ -23,3 +30,29 @@ def delete(self, request):
         logout(request)
         logger.info("User '%s' with email '%s' has been deleted.", user.username, user.email)
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+
+
+class UserDataDumpView(APIView):
+    permission_classes = [IsAuthenticated]
+
+    def get(self, request):
+        response = HttpResponse(content_type='text/csv')
+        response['Content-Disposition'] = 'attachment; filename="export.csv"'
+        writer = csv.DictWriter(response, fieldnames=['video_a', 'video_b', 'criteria', 'weight', 'score'])
+        writer.writeheader()
+                
+        comparison_queryset = Comparison.objects.filter(user=request.user)
```

Comment:
Suggestion to reduce the number of potential SQL queries required to build the export:
```suggestion
        comparison_queryset = Comparison.objects\
            .filter(user=request.user)\
            .select_related("video_1", "video_2")\
            .prefetch_related("criteria_scores")

```

Reply:
Thanks for the optimization! Made the change as you recommend